p2p/conn/connection.go

```diff
@@ -664,6 +704,31 @@ FOR_LOOP:
 	}
 }
 
+func (c *MConnection) recvMsgRoutine() {
+	defer c._recover()
+
+	for {
+		connMsg, ok := <-c.msgRecvQueue
+
+		if !ok {
+			break
+		}
+
+		msgLen := len(connMsg.msgBytes)
+		// Will never block as long as msgRecvAck is at least one bigger than msgRecvQueue
+		c.msgRecvAck <- uint32(msgLen)
+		// The following logging of c.msgRecvSize is not a correct representation of the current aggregate size
+		// of the queue but is instead the size of the queue as accounted by the producer routine (trailing behind)
```

Comment:
"producer routine"?

Reply:
The producer for this channel is `recvRoutine`. I admit we could be better at names here

Reply:
Should we update the names?

Reply:
I wanted to avoid changing the name of the original routine to keep future conflict resolution to a minimum, but maybe that's a misplaced hope

Reply:
ok !!